File: backend/elasticpom/src/main/java/com/elasticpom/core/model/Paper.java
package com.elasticpom.core.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class Paper {
    private String id;
    private String paperId;
    private LocalDate datestamp;
    private List<String> creators;
    private List<String> subjects;
    private String description;
    private String publisher;
    private List<String> contributors;
    private LocalDate date;
    private String type;
    private String format;
    private String identifier;
    private String source;
    private String language;
    private List<String> relations;
    private String coverage;
    private String rights;
    private String paperType;
    private List<UniqueField> uniqueFields;
}

File: frontend/src/lib/types/paper.ts
export interface UniqueField {
	name: string;
	data: string;
}

export interface PaperDto {
	id: string;
	paperId: string;
	datestamp: string | null;
	title: string;
	creators: string[];
	subjects: string[];
	description: string;
	publisher: string;
	contributors: string[];
	date: string | null;
	type: string;
	format: string;
	identifier: string;
	source: string;
	language: string;
	relations: string[];
	coverage: string;
	rights: string;
	paperType: string;
	uniqueFields: UniqueField[];
}
